python/orca/src/bigdl/orca/learn/tf2/spark_runner.py

```diff
@@ -265,15 +265,14 @@ def __init__(self,
                         else:
                             if self.model_creator is not None:
                                 self.model = self.model_creator(self.config)
-                                if self.optimizer_weights:
-                                    def load_opt_weights():
+                                if self.model_variables:
+                                    def build_optimizer():
                                         grad_vars = self.model.trainable_weights
                                         zero_grads = [tf.zeros_like(w) for w in grad_vars]
                                         self.model.optimizer.apply_gradients(
                                             zip(zero_grads, grad_vars))
-                                        self.model.optimizer.set_weights(
-                                            self.optimizer_weights.value)
-                                    self.strategy.run(load_opt_weights)
+                                        self.model.optimizer.build(self.model_variables)
```

Comment:
is this compatible with older tf versions?

Reply:
nope, fixing this part.